finagle-core/src/test/scala/com/twitter/finagle/filter/RequestMeterFilterTest.scala

```diff
@@ -0,0 +1,84 @@
+package com.twitter.finagle.filter
+
+import java.util.concurrent.RejectedExecutionException
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.conversions.time._
+import com.twitter.finagle.{Failure, Service}
+import com.twitter.util._
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.mockito.Mockito.when
+
+@RunWith(classOf[JUnitRunner])
+class RequestMeterFilterTest extends FunSuite with MockitoSugar {
+
+  val echoSvc = new Service[Int, Int] {
+    def apply(req: Int) = Future(req)
+  }
+
+  test("return service execution after getting a permit") {
+    val timer = new MockTimer
+    Time.withCurrentTimeFrozen { ctl =>
+      val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
+      val svc = new RequestMeterFilter(meter) andThen echoSvc
+
+      assert(Await.result(svc(1)) == 1)
+    }
+  }
+
+  test("mark dropped requests as failed") {
+    val timer = new MockTimer
+    Time.withCurrentTimeFrozen { ctl =>
+      val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
+      val svc = new RequestMeterFilter(meter) andThen echoSvc
+
+      val f1 = svc(1)
+      assert(f1.isDefined)
+
+      val f2 = svc(2)
+      assert(!f2.isDefined)
+
+      val f3 = svc(3)
+      assert(f3.isDefined)
+      val failure = intercept[Failure] { Await.result(f3, 5.seconds) }
+      assert(failure.getCause.getClass == classOf[RejectedExecutionException])
```

Comment:
Quite liked it! Was looking for a cleaner way of expressing this assertion! Thanks!